### PR TITLE
Issues #37 and #16: Added multi-cluster support and added cluster-name in botkube commands

### PR DESCRIPTION
**Pull request by mugdha-adhav**
Added cluster name in the notifier, start, stop and botkube commands.

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -22,11 +22,6 @@ import (
 
 var startTime time.Time
 
-const (
-	controllerStartMsg = "...and now my watch begins! :crossed_swords:"
-	controllerStopMsg  = "my watch has ended!"
-)
-
```
Comment (PrasadG193) [inline pkg/controller/controller.go]: Don't remove constants. Instead, use formatting methods like fmt.Sprintf() for string manipulations

Comment (mugdha-adhav) [reply pkg/controller/controller.go]: Added constants and used Sprintf

File: design/multi-cluster.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+To get the list of all clusters configured in the slack, you can run the following command in slack.
+
+```sh
+@Botkube ping
+```
+
+##### Workflow
+
+![alt text](https://raw.githubusercontent.com/mugdha-adhav/botkube/infracloudio/add-cluster-name/design/workflow.png)
```
Comment (ssudake21) [inline design/multi-cluster.md]: add image in design folder itself. Make this link relative

Comment (mugdha-adhav) [reply design/multi-cluster.md]: Yes got that. Updating in next commit.

File: pkg/execute/executor.go [hunk 3]
```diff
@@ -103,71 +119,102 @@ func printDefaultMsg() string {
...
-		}
-		if args[1] == "start" {
-			config.Notify = true
-			log.Logger.Info("Notifier enabled")
-			return notifierStartMsg
+func runNotifierCommand(args []string, clustername string, messagetype string) string {
+	if messagetype != "authChannelMsg" {
+		return ""
+	}
+	if args[1] == "start" {
```
Comment (ssudake21) [inline pkg/execute/executor.go]: Can we please define some constants for these commands ?
start,stop, status, etc

Comment (mugdha-adhav) [reply pkg/execute/executor.go]: Removed messageType variable and using bool value instead. Will update in the next commit.

File: pkg/execute/executor.go [hunk 4]
```diff
@@ -103,71 +119,102 @@ func printDefaultMsg() string {
...
 		if a == "-w" || strings.HasPrefix(a, "--watch") {
 			continue
 		}
+		if regexp.MustCompile("--cluster-name(.*)").MatchString(a) {
+			if a == "--cluster-name" {
+				checkFlag = true
+			} else if strings.SplitAfterN(a, "--cluster-name=", 2)[1] != clustername {
+				return ""
+			}
+			messagetype = "authChannelMsg"
```
[the remark this answers is not in the document]
Comment (mugdha-adhav) [reply pkg/execute/executor.go]: Removed messageType variable and using bool value instead. Will update in the next commit.

File: pkg/execute/executor.go [hunk 5]
```diff
@@ -64,37 +106,46 @@ func (e *DefaultExecutor) Execute() string {
...
 	return unsupportedCmdMsg
 }
 
-func printHelp() string {
+func printHelp(channelName string) string {
 	allowedKubectl := ""
 	for k := range validKubectlCommands {
 		allowedKubectl = allowedKubectl + k + ", "
 	}
 	helpMsg := "BotKube executes kubectl commands on k8s cluster and returns output.\n" +
```
Comment (ssudake21) [inline pkg/execute/executor.go]: just use a multiline string here & format with sprintf
https://play.golang.org/p/BxJj5R-qspK

File: design/multi-cluster.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+To get the list of all clusters configured in the slack, you can run the following command in slack.
+
+```sh
+@Botkube ping
+```
+
+##### Workflow
+
+![Multi_Cluster_Design](workflow.md)
```
Comment (ssudake21) [inline design/multi-cluster.md]: whats workflow.md ?

Comment (ssudake21) [reply design/multi-cluster.md]: please make sure image is appearing correctly here https://github.com/infracloudio/botkube/blob/infracloudio/add-cluster-name/design/multi-cluster.md

File: pkg/execute/executor.go (hunk 5 above)
Comment (mugdha-adhav) [reply pkg/execute/executor.go]: Made required changes.

File: design/multi-cluster.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+To get the list of all clusters configured in the slack, you can run the following command in slack.
+
+```sh
+@Botkube ping
+```
+
+##### Workflow
+
+![Multi_Cluster_Design](https://raw.githubusercontent.com/infracloudio/botkube/infracloudio/add-cluster-name/design/workflow.png)
```
[the remark this answers is not in the document]
Comment (mugdha-adhav) [reply design/multi-cluster.md]: Made required changes.

**Pull request merged**